### PR TITLE
ref. builder: improve arxiv recognition

### PR DESCRIPTION
**Pull request by MJedr**
Ref: inspirehep/inspirehep#1208

### CODE REVIEW

File: inspire_schemas/builders/references.py [hunk 1]
```diff
@@ -200,11 +200,14 @@ def set_year(self, year):
...
-            self._add_uid(url, True)
-        except ValueError:
+    def add_url(self, url, is_not_uid=False):
+        if is_not_uid:
             self._add_url(url)
+        else:
+            try:
+                self._add_uid(url, True)
+            except ValueError:
+                self._add_url(url)
```
Comment (michamos) [inline inspire_schemas/builders/references.py:210]: isn't this whole block now equivalent to `self._add_url(url)`? what is the point of this change?

File: inspire_schemas/utils.py [hunk 2]
```diff
@@ -87,12 +87,12 @@
 # Matches new style arXiv ID, with an old-style class specification
 # (Malformed, but appears in APS records)
 RE_ARXIV_POST_2007_CLASS = re.compile(
-    r"(arxiv:)?((?P<category>(?:[a-z-]+)(?:\.[a-z]{2})?)/)?(?P<identifier>\d{4}\.\d{4,5})(v\d+)?$",
+    r"(arxiv:)?((?P<category>(?:[a-z-]+)(?:\.[a-z]{2})?)/)?(?P<identifier>\d{4}\.\d{4,5})(v\d+)?(\[[a-z|\-|\.]*\])?$",
```
Comment (michamos) [inline inspire_schemas/utils.py:90]: You can use the same pattern as for `category` to make it stricter. Also maybe allow whitespace before the `[`

File: inspire_schemas/builders/references.py (hunk 1 above)
Comment (MJedr) [reply inspire_schemas/builders/references.py:210]: It is, but in theory this is a protected method so I can't use it from outside of the app

File: inspire_schemas/utils.py [hunk 3]
```diff
@@ -87,12 +87,12 @@
 # Matches new style arXiv ID, with an old-style class specification
 # (Malformed, but appears in APS records)
 RE_ARXIV_POST_2007_CLASS = re.compile(
-    r"(arxiv:)?((?P<category>(?:[a-z-]+)(?:\.[a-z]{2})?)/)?(?P<identifier>\d{4}\.\d{4,5})(v\d+)?$",
+    r"(arxiv:)?((?P<category>(?:[a-z-]+)(?:\.[a-z]{2})?)/)?(?P<identifier>\d{4}\.\d{4,5})(v\d+)?\ ?(\[(?:[a-z-]+)(?:\.[a-z]{2})\])?$",
```
Comment (michamos) [inline inspire_schemas/utils.py:90]:
```suggestion
    r"(arxiv:)?((?P<category>(?:[a-z-]+)(?:\.[a-z]{2})?)/)?(?P<identifier>\d{4}\.\d{4,5})(v\d+)?\s*(\[(?:[a-z-]+)(?:\.[a-z]{2})\])?$",
```

File: inspire_schemas/utils.py [hunk 4]
```diff
@@ -87,12 +87,12 @@
...
     flags=re.I
 )
 
 RE_ARXIV_PRE_2007_CLASS = re.compile(
-    r"(arxiv:)?(?P<category>(?P<extraidentifier>[a-z-]+)(?:\.[a-z]{2})?)/(?P<identifier>\d{4}\d+)(v\d+)?$",
+    r"(arxiv:)?(?P<category>(?P<extraidentifier>[a-z-]+)(?:\.[a-z]{2})?)/(?P<identifier>\d{4}\d+)(v\d+)?\ ?(\[[a-z-]+)(?:\.[a-z]{2}\])?$",
```
Comment (michamos) [inline inspire_schemas/utils.py:95]:
```suggestion
    r"(arxiv:)?(?P<category>(?P<extraidentifier>[a-z-]+)(?:\.[a-z]{2})?)/(?P<identifier>\d{4}\d+)(v\d+)?\s*(\[[a-z-]+)(?:\.[a-z]{2}\])?$",
```

**Review by michamos — APPROVED** [review]
:+1: after the small change

**Pull request merged**